presto-parser/src/main/java/com/facebook/presto/sql/tree/TableProperties.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.util.Objects.requireNonNull;
+
+public class TableProperties
+        extends Node
+{
+    private final List<TableProperty> properties;
+
+    public TableProperties()
+    {
+        this(ImmutableList.of());
+    }
+
+    public TableProperties(List<TableProperty> properties)
+    {
+        this(Optional.empty(), properties);
+    }
+
+    public TableProperties(NodeLocation location, List<TableProperty> properties)
+    {
+        this(Optional.of(location), properties);
+    }
+
+    private TableProperties(Optional<NodeLocation> location, List<TableProperty> properties)
+    {
+        super(location);
+        requireNonNull(properties, "properties is null");
+        this.properties = properties.stream()
+                .sorted(Comparator.comparing(property -> property.getName().getValue()))
+                .collect(toImmutableList());
+    }
+
+    public List<TableProperty> getProperties()
+    {
+        return properties;
+    }
+
+    public Map<String, Expression> buildPropertyMap()
```

Comment:
This shouldn't be part of the node. It needs to take into account the rules for determining whether two identifiers are equivalent and raise an appropriate semantic error if there are duplicates. This logic belongs in the analysis phase.